src/tests/java/htsjdk/samtools/sra/AbstractSRATest.java

```diff
@@ -0,0 +1,56 @@
+package htsjdk.samtools.sra;
+
+import htsjdk.samtools.SAMRecord;
+import htsjdk.samtools.SAMRecordIterator;
+import org.testng.Assert;
+import org.testng.SkipException;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.util.NoSuchElementException;
+
+@Test(groups = "SRA")
+public abstract class AbstractSRATest {
+
+    @BeforeMethod
+    public final void assertSRAIsSupported(){
+        if(!SRAAccession.isSupported()){
+            throw new SkipException("Skipping SRA Test because SRA native code is unavailable.");
+        }
+    }
+
+    protected static void checkAlignedUnalignedCountsByIterator(SAMRecordIterator samRecordIterator,
```

Comment:
Could you rename this while we are making changes?  Perhaps `checkMappedStatusCounts`?